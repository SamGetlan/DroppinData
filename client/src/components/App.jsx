import React from 'react';
import axios from 'axios';
import Navbar from './Navbar.jsx';
import Body from './Body.jsx';
import UserForm from './UserForm.jsx';
import FilterLocations from './FilterLocations.jsx';
import AccountOptionsForm from './AccountOptionsForm.jsx';
import FullMap from './FullMap.jsx';
import locations from '../data.js';


class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      focus: '',
      locations,
      filteredLocations: locations.slice(),
      userFormActive: true,
      filterLocationsActive: false,
      signUpForm: false,
      loggedIn: false,
      active: false,
      activeIndex: false,
      userGames: null,
      submitButtonState: true,
      accountOptionsForm: false,
      logInFailed: null,
      userGameData: null,
      showFullMap: false,
    };
    this.handleUserFormClick = this.handleUserFormClick.bind(this);
    this.handleFilterClick = this.handleFilterClick.bind(this);
    this.loginUserFormOption = this.loginUserFormOption.bind(this);
    this.signUpUserFormOption = this.signUpUserFormOption.bind(this);
    this.handleAccountSignUp = this.handleAccountSignUp.bind(this);
    this.handleAccountSignIn = this.handleAccountSignIn.bind(this);
    this.filterOut = this.filterOut.bind(this);
    this.filterIn = this.filterIn.bind(this);
    this.handleGameSubmit = this.handleGameSubmit.bind(this);
    this.handleActionClick = this.handleActionClick.bind(this);
    this.handleTileClick = this.handleTileClick.bind(this);
    this.handleAccountOptionsClick = this.handleAccountOptionsClick.bind(this);
    this.handleLogout = this.handleLogout.bind(this);
    this.getUserGameData = this.getUserGameData.bind(this);
    this.hardGroupClick = this.hardGroupClick.bind(this);
    this.notRecentGroupClick = this.notRecentGroupClick.bind(this);
    this.killsGroupClick = this.killsGroupClick.bind(this);
    this.placeGroupClick = this.placeGroupClick.bind(this);
    this.popularGroupClick = this.popularGroupClick.bind(this);
    this.filterAllIn = this.filterAllIn.bind(this);
    this.filterAllOut = this.filterAllOut.bind(this);
    this.handleShowMapClick = this.handleShowMapClick.bind(this);
    this.handleMapChoiceClick = this.handleMapChoiceClick.bind(this);
  }

  handleUserFormClick() {
    this.setState({
      userFormActive: !this.state.userFormActive,
    });
  }

  handleFilterClick() {
    this.setState({
      filterLocationsActive: !this.state.filterLocationsActive,
    });
  }

  handleActionClick() {
    const { filteredLocations } = this.state;
    const index = Math.floor(Math.random() * filteredLocations.length);
    console.log(filteredLocations[index].camelCase);
    this.setState({
      active: filteredLocations[index].camelCase,
      activeIndex: index,
    });
  }

  loginUserFormOption() {
    this.setState({
      signUpForm: false,
    });
  }

  signUpUserFormOption() {
    this.setState({
      signUpForm: true,
    });
  }

  handleAccountSignUp(username, password) {
    const context = this;
    axios.post('/api/signup', {
      username,
      password,
      createdAt: new Date(),
    })
      .then((data) => {
        console.log('signUp successful:', data.config.data);
        const user = JSON.parse(data.config.data).username;
        context.setState({
          loggedIn: user,
          userFormActive: false,
        });
      })
      .catch((err) => {
        console.log('There was an error:', err);
      });
  }

  handleAccountSignIn(username, password) {
    const context = this;
    axios.post('/api/login', {
      username,
      password,
    })
      .then((data) => {
        if (data.data.username) {
          console.log('Login successful:', data);
          const user = (data.data.username);
          context.setState({
            loggedIn: user,
            userFormActive: false,
            logInFailed: false,
          });
          axios.get('/api/games')
            .then((data) => {
              console.log('data received -->', data);
              const userGameData = context.getUserGameData(data.data);
              context.setState({
                userGames: data.data,
                userGameData,
              });
            });
        } else {
          console.log('Login failed:', data.data);
          context.setState({
            logInFailed: true,
          });
        }
      })
      .catch((err) => {
        console.log('There was an error:', err);
      });
  }

  filterIn(location) {
    console.log(this.state.filteredLocations);
    if (this.state.filteredLocations.indexOf(location) === -1) {
      this.state.filteredLocations.push(location);
      const filteredLocations = this.state.filteredLocations;
      this.setState({ filteredLocations });
    }
  }

  filterOut(location) {
    console.log(this.state.filteredLocations);
    const index = this.state.filteredLocations.indexOf(location);
    if (index !== -1) {
      this.state.filteredLocations.splice(index, 1);
      const filteredLocations = this.state.filteredLocations;
      this.setState({ filteredLocations });
    }
  }

  handleGameSubmit(place, kills, loot, gameType, death) {
    const context = this;
    this.setState({
      submitButtonState: false,
    })
    // console.log('Inside the handleSubmit function');
    // console.log(`place --> ${place}`);
    // console.log(`kills --> ${kills}`);
    // console.log('loot -->', loot);
    // console.log('death -->', death);
    // console.log(`Game Type --> ${gameType}`);
    if (place >= 1 && place <= 100 && kills <= 100 && kills >= 0 && gameType && this.state.loggedIn && death !== 'null') {
      axios.post('/api/games', {
        user: this.state.loggedIn,
        date: new Date(),
        location: this.state.filteredLocations[this.state.activeIndex].name,
        place,
        kills,
        loot,
        gameType,
        death,
      })
        .then((data) => {
          console.log('We have received Data -->', data);
          context.setState({
            userGames: data.data,
          });
          const $confirm = document.createElement('p');
          $confirm.innerHTML = 'Game was successfully submitted';
          document.getElementsByClassName('statBox')[0].appendChild($confirm);
          setTimeout(() => {
            context.setState({
              submitButtonState: true,
            })
          }, 1000);
          setTimeout(() => {
            $confirm.remove();
          }, 4000);
        })
        .catch((err) => {
          console.log('There was an Error -->', err);
        });
    } else {
      let suggestion;
      if (place < 1 || place > 100 || place === undefined) {
        suggestion = 'Your place needs to be between 1 and 100.';
      } else if (kills < 0 || kills > 100 || kills === undefined) {
        suggestion = 'Your kills need to be between 0 and 100.';
      } else if (loot < 1 || loot > 10) {
        suggestion = 'Loot rating must be between 1 and 10.';
      } else if (!gameType) {
        suggestion = 'A game type must be selected.';
      } else if (death === 'null') {
        suggestion = 'A death location must be selected.';
      } else if (!this.state.loggedIn) {
        suggestion = 'You must log in to submit a game.';
      }
      const $denied = document.createElement('p');
      $denied.innerHTML = `There was an error submitting your game. ${suggestion}`;
      document.getElementsByClassName('statBox')[0].appendChild($denied);
      setTimeout(() => {
        context.setState({
          submitButtonState: true,
        })
      }, 1000);
      setTimeout(() => {
        $denied.remove();
      }, 4000);
    }
  }

  handleTileClick(location) {
    console.log('inside handleTileClick:', location);
    const index = this.state.filteredLocations.findIndex(element => (element.camelCase === location.camelCase));
    console.log('active Location -->', location);
    console.log('active index -->', index);
    this.setState({
      active: location.camelCase,
      activeIndex: index,
    });
    document.body.scrollTop = 0; // For Safari
    document.documentElement.scrollTop = 0; // For everything else
  }

  handleAccountOptionsClick() {
    console.log('inside handleAccountOptionsClick');
    this.setState({
      accountOptionsForm: !this.state.accountOptionsForm,
    });
  }

  handleLogout() {
    const context = this;
    axios.get('/api/logout')
      .then((data) => {
        context.setState({
          loggedIn: false,
          accountOptionsForm: false,
          userGames: null,
        })
      })
      .catch((err) => {
        console.log('err:', err);
      })
  }

  getUserGameData(games) {
    const results = {};
    const getPlace = (game) => {
      if (game.gameType === 'solo') {
        return game.place;
      } else if (game.gameType === 'duo') {
        return game.place * 2;
      } else if (game.gameType === 'squad') {
        return game.place * 4;
      }
    }
    const updateStats = (results, game, gameType) => {
      results[game.location][gameType].totalKills += game.kills;
      results[game.location][gameType].totalPlace += (gameType === 'all' ? getPlace(game) : game.place);
      results[game.location][gameType].totalLoot += game.loot;
      results[game.location][gameType].bestKills = Math.max(results[game.location][gameType].bestKills, game.kills);
      results[game.location][gameType].bestPlace = Math.min(results[game.location][gameType].bestPlace, (gameType === 'all' ? getPlace(game) : game.place));
      results[game.location][gameType].bestLoot = Math.max(results[game.location][gameType].bestLoot, game.loot);
      results[game.location][gameType].recentKills = game.kills;
      results[game.location][gameType].recentPlace = (gameType === 'all' ? getPlace(game) : game.place);
      results[game.location][gameType].recentLoot = game.loot;
      results[game.location][gameType].totalGames++;
      if (game.date > results[game.location][gameType].mostRecent) {
        results[game.location][gameType].mostRecent = game.date;
      }
    }
    games.forEach((game, index, games) => {
      if (!results[game.location]) {
        results[game.location] = {
          solo: {
            averageKills: 0,
            averagePlace: 0,
            averageLoot: 0,
            bestKills: 0,
            bestPlace: 101,
            bestLoot: 0,
            totalKills: 0,
            totalPlace: 0,
            totalLoot: 0,
            totalGames: 0,
            recentKills: 0,
            recentPlace: 0,
            recentLoot: 0,
            mostRecent: game.date,
          },
          duo: {
            averageKills: 0,
            averagePlace: 0,
            averageLoot: 0,
            bestKills: 0,
            bestPlace: 101,
            bestLoot: 0,
            totalKills: 0,
            totalPlace: 0,
            totalLoot: 0,
            totalGames: 0,
            recentKills: 0,
            recentPlace: 0,
            recentLoot: 0,
            mostRecent: game.date,
          },
          squad: {
            averageKills: 0,
            averagePlace: 0,
            averageLoot: 0,
            bestKills: 0,
            bestPlace: 101,
            bestLoot: 0,
            totalKills: 0,
            totalPlace: 0,
            totalLoot: 0,
            totalGames: 0,
            recentKills: 0,
            recentPlace: 0,
            recentLoot: 0,
            mostRecent: game.date,
          },
          all: {
            averageKills: 0,
            averagePlace: 0,
            averageLoot: 0,
            bestKills: 0,
            bestPlace: 101,
            bestLoot: 0,
            totalKills: 0,
            totalPlace: 0,
            totalLoot: 0,
            totalGames: 0,
            recentKills: 0,
            recentPlace: 0,
            recentLoot: 0,
            mostRecent: game.date,
          },
        }
      }
      updateStats(results, game, 'all');
      updateStats(results, game, game.gameType);
    });
    for (var location in results) {
      results[location].solo.averageKills = results[location].solo.totalKills / results[location].solo.totalGames;
      results[location].duo.averageKills = results[location].duo.totalKills / results[location].duo.totalGames;
      results[location].squad.averageKills = results[location].squad.totalKills / results[location].squad.totalGames;
      results[location].all.averageKills = results[location].all.totalKills / results[location].all.totalGames;
      results[location].solo.averagePlace = results[location].solo.totalPlace / results[location].solo.totalGames;
      results[location].duo.averagePlace = results[location].duo.totalPlace / results[location].duo.totalGames;
      results[location].squad.averagePlace = results[location].squad.totalPlace / results[location].squad.totalGames;
      results[location].all.averagePlace = results[location].all.totalPlace / results[location].all.totalGames;
    }
    return results;
  }

  hardGroupClick() {
    const type = 'all';
    if (this.state.userGames.length > 9) {
      const results = [];
      const userGameData = this.state.userGameData;
      const hardGroup = Object.keys(this.state.userGameData);
      hardGroup.sort((a, b) => {
        return userGameData[b][type].averagePlace - userGameData[a][type].averagePlace;
      });
      for (var i = 0; i < Math.min(hardGroup.length, 8); i++) {
        for (var ii = 0; ii < locations.length; ii++) {
          if (locations[ii].name === hardGroup[i]) {
            results.push(locations[ii]);
          }
        }
      }
      this.setState({
        filteredLocations: results,
      });

    } else {
      console.log('Need to save at least 10 games in order for customSorting to work');
    }
  }

  notRecentGroupClick(gameType = 'all') {
    console.log('inside notRecentGroupClick');
    const type = 'all';
    if (this.state.userGames.length > 9) {
      const results = [];
      const userGameData = this.state.userGameData;
      const notRecentGroup = Object.keys(this.state.userGameData);
      notRecentGroup.sort((a, b) => {
        return userGameData[a][type].mostRecent - userGameData[b][type].mostRecent;
      });
      for (var i = 0; i < Math.min(notRecentGroup.length, 8); i++) {
        for (var ii = 0; ii < locations.length; ii++) {
          if (locations[ii].name === notRecentGroup[i]) {
            results.push(locations[ii]);
          }
        }
      }
      this.setState({
        filteredLocations: results,
      });
    }
  }

  killsGroupClick() {
    console.log('inside killsGroupClick');
    const type = 'all';
    if (this.state.userGames.length > 9) {
      const results = [];
      const userGameData = this.state.userGameData;
      const killsGroup = Object.keys(this.state.userGameData);
      killsGroup.sort((a, b) => {
        return userGameData[b][type].averageKills - userGameData[a][type].averageKills;
      });
      for (var i = 0; i < Math.min(killsGroup.length, 8); i++) {
        for (var ii = 0; ii < locations.length; ii++) {
          if (locations[ii].name === killsGroup[i]) {
            results.push(locations[ii]);
          }
        }
      }
      this.setState({
        filteredLocations: results,
      });
    } else {
      console.log('Need to save at least 10 games in order for customSorting to work');
    }
  }

  placeGroupClick() {
    console.log('inside placeGroupClick');
    const type = 'all';
    if (this.state.userGames.length > 9) {
      const results = [];
      const userGameData = this.state.userGameData;
      const easyGroup = Object.keys(this.state.userGameData);
      easyGroup.sort((a, b) => {
        return userGameData[a][type].averagePlace - userGameData[b][type].averagePlace;
      });
      for (var i = 0; i < Math.min(easyGroup.length, 8); i++) {
        for (var ii = 0; ii < locations.length; ii++) {
          if (locations[ii].name === easyGroup[i]) {
            results.push(locations[ii]);
          }
        }
      }
      this.setState({
        filteredLocations: results,
      });

    } else {
      console.log('Need to save at least 10 games in order for customSorting to work');
    }
  }

  popularGroupClick() {
    console.log('inside popularGroupClick');
    const type = 'all';
    if (this.state.userGames.length > 9) {
      const results = [];
      const userGameData = this.state.userGameData;
      const popularGroup = Object.keys(this.state.userGameData);
      popularGroup.sort((a, b) => {
        return userGameData[b][type].totalGames - userGameData[a][type].totalGames;
      });
      for (var i = 0; i < Math.min(popularGroup.length, 8); i++) {
        for (var ii = 0; ii < locations.length; ii++) {
          if (locations[ii].name === popularGroup[i]) {
            results.push(locations[ii]);
          }
        }
      }
      this.setState({
        filteredLocations: results,
      });

    } else {
      console.log('Need to save at least 10 games in order for customSorting to work');
    }
  }

  filterAllIn() {
    this.setState({
      filteredLocations: locations.slice(),
    });
  }

  filterAllOut() {
    console.log('inside filterAllOut');
    this.setState({
      filteredLocations: [],
    });
  }

  handleShowMapClick() {
    this.setState({
      showFullMap: !this.state.showFullMap,
    });
  }

  handleMapChoiceClick(e) {
    console.log('inside handleMapChoiceClick from div -->', e.target.id);
    for (var i = 0; i < locations.length; i++) {
      if (locations[i].camelCase === e.target.id) {
        this.filterIn(locations[i]);
      }
    }
    for (var k = 0; k < this.state.filteredLocations.length; k++) {
      if (this.state.filteredLocations[k].camelCase === e.target.id) {
        this.setState({
          active: e.target.id,
          activeIndex: k,
          showFullMap: false,
        });
      }
    }
  }

  render() {
    return (
      <div id="app">
        <Navbar
          navButtons={['Home', 'Map', 'Filter Locations', 'Sign Up or Login']}
          classes={['home', 'map', 'filter', 'login']}
          handleUserFormClick={this.handleUserFormClick}
          handleFilterClick={this.handleFilterClick}
          loggedIn={this.state.loggedIn}
          handleAccountOptionsClick={this.handleAccountOptionsClick}
          handleShowMapClick={this.handleShowMapClick}
        />
        <Body
          filteredLocations={this.state.filteredLocations}
          activeIndex={this.state.activeIndex}
          handleSubmit={this.handleGameSubmit}
          active={this.state.active}
          handleActionClick={this.handleActionClick}
          userGames={this.state.userGames}
          handleTileClick={this.handleTileClick}
          submitButtonState={this.state.submitButtonState}
          locations={this.state.locations}
        />
        {this.state.userFormActive &&
        <UserForm
          handleUserFormClick={this.handleUserFormClick}
          signUpForm={this.state.signUpForm}
          loginUserFormOption={this.loginUserFormOption}
          signUpUserFormOption={this.signUpUserFormOption}
          handleAccountSignIn={this.handleAccountSignIn}
          handleAccountSignUp={this.handleAccountSignUp}
          logInFailed={this.state.logInFailed}
        />}
        {this.state.filterLocationsActive &&
        <FilterLocations
          locations={this.state.locations}
          filteredLocations={this.state.filteredLocations}
          filterOut={this.filterOut}
          filterIn={this.filterIn}
          handleFilterClick={this.handleFilterClick}
          hardGroupClick={this.hardGroupClick}
          notRecentGroupClick={this.notRecentGroupClick}
          killsGroupClick={this.killsGroupClick}
          placeGroupClick={this.placeGroupClick}
          popularGroupClick={this.popularGroupClick}
          filterAllIn={this.filterAllIn}
          filterAllOut={this.filterAllOut}
        />}
        {this.state.accountOptionsForm &&
        <AccountOptionsForm
          handleAccountOptionsClick={this.handleAccountOptionsClick}
          loggedIn={this.state.loggedIn}
          handleLogout={this.handleLogout}
        />}
        {this.state.showFullMap &&
        <FullMap
          handleShowMapClick={this.handleShowMapClick}
          locations={this.state.locations}
          handleMapChoiceClick={this.handleMapChoiceClick}
        />}
      </div>
    );
  }
}

export default App;
