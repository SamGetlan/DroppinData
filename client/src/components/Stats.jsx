import React from 'react';
import { Route } from 'react-router-dom';
import Navbar from './Navbar.jsx';
import FilterMyGames from './FilterMyGames.jsx';
import MyGames from './MyGames.jsx';
import StatDashboard from './StatDashboard.jsx';


class Stats extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div id="StatsPageContainer">
        <Navbar
          navButtons={['Home', 'Dashboard', 'My Games', 'Sign Up or Login']}
          classes={['home', 'dashboard', 'myGames', 'login']}
          // handleUserFormClick={this.props.handleUserFormClick}
          loggedIn={this.props.loggedIn}
          handleAccountOptionsClick={this.props.handleAccountOptionsClick}
        />
        <FilterMyGames 
          handleFiltering={this.props.handleFiltering}
          filterOptions={this.props.filterOptions}
          locations={this.props.locations}
        />
        <Route path="/stats/myGames" render={props => <MyGames {...props}
          userGames={this.props.userGames}
          filteredUserGames={this.props.filteredUserGames}
          confirmDeleteGameCard={this.props.confirmDeleteGameCard}
          locations={this.props.locations}
          updateLocalGame={this.props.updateLocalGame}
          handleNotCompliantEditGameSubmission={this.props.handleNotCompliantEditGameSubmission}
          statLoading={this.props.statLoading}
        />} />
        <Route path="/stats/dashboard" render={props => <StatDashboard {...props}
          userGames={this.props.userGames}
          filteredUserGames={this.props.filteredUserGames}
          data={this.props.dashboardData}
          statLoading={this.props.statLoading}
          getDashboardData={this.props.getDashboardData}
          getPieChartData={this.props.getPieChartData}
          pieChartData={this.props.pieChartData}
        />} />
      </div>
    );
  }
}

export default Stats;