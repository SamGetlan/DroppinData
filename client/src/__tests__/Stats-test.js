import React from 'react';
import renderer from 'react-test-renderer';
import App from '../components/App.jsx';
import Stats from '../components/Stats.jsx';
import { MemoryRouter } from 'react-router';
import locations from '../data.js';

test('Stats renders correctly', () => {
  const element = renderer.create(
    <MemoryRouter>
      <Stats
        userGames={null}
        navButtons={['Home', 'Dashboard', 'My Games', 'Sign Up or Login']}
        classes={['home', 'dashboard', 'myGames', 'login']}
        handleUserFormClick={App.handleUserFormClick}
        loggedIn={false}
        handleAccountOptionsClick={App.handleAccountOptionsClick}
        confirmDeleteGameCard={App.confirmDeleteGameCard}
        filteredUserGames={null}
        filterOptions={{
          startLocation: 'All',
          deathLocation: 'All',
          worstPlace: 100,
          bestPlace: 1,
          worstKills: 0,
          bestKills: 99,
          worstLoot: 0,
          bestLoot: 10,
          minDistanceTraveled: 0,
          maxDistanceTraveled: 3640,
          days: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
          timeStart: '00:00',
          timeEnd: '23:59',
        }}
        updateFilteredUserGames={App.updateFilteredUserGames}
        locations={locations}
        updateLocalGame={App.updateLocalGame}
        handleFiltering={App.handleFiltering}
        handleNotCompliantEditGameSubmission={App.handleNotCompliantEditGameSubmission}
        dashboardData={null}
        statLoading={false}
        getDashboardData={App.getDashboardData}
        getPieChartData={App.getPieChartData}
        pieChartData={null}
      />
    </MemoryRouter>
  );
  let tree = element.toJSON();
  expect(tree).toMatchSnapshot();
});