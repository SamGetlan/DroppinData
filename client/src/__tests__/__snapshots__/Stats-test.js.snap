// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats renders correctly 1`] = `
<div
  id="StatsPageContainer"
>
  <div
    className="navbar row"
  >
    <a
      className="navButtonContainer"
      href="/home"
      onClick={[Function]}
    >
      <button
        className="navButton col-2 col-s-6 home"
      >
        Home
      </button>
    </a>
    <a
      className="navButtonContainer"
      href="/stats/dashboard"
      onClick={[Function]}
    >
      <button
        className="navButton col-2 col-s-6 dashboard"
      >
        Dashboard
      </button>
    </a>
    <a
      className="navButtonContainer"
      href="/stats/myGames"
      onClick={[Function]}
    >
      <button
        className="navButton col-2 col-s-6 myGames"
      >
        My Games
      </button>
    </a>
    <a
      className="navButtonContainer"
      href="/home/login"
      onClick={[Function]}
    >
      <button
        className="navButton col-3 col-s-6 login"
      >
        Sign Up or Login
      </button>
    </a>
  </div>
  <div
    className="filterMyGamesContainer"
  >
    <div
      className="filterOptionsBar"
    >
      <div
        className="filterOption"
      >
        <label>
          Start Location
        </label>
        <div
          className="MuiInput-root-6 MuiInput-underline-10"
        >
          <div
            className="MuiSelect-root-1"
          >
            <div
              aria-haspopup="true"
              aria-owns={null}
              aria-pressed="false"
              className="MuiSelect-select-2 MuiSelect-selectMenu-3 MuiInput-input-14"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              All
            </div>
            <input
              type="hidden"
              value="Chair,Loot Lake House,West of Loot Lake,North of Loot Lake,Prison,Shipping Containers,Lucky Landing,Flush Factory,Tilted Towers,Shifty Shafts,Haunted Hills,Anarchy Acres,Lonely Lodge,Wailing Woods,Moisty Mire,Fatal Fields,Greasy Grove,Snobby Shores,Junk Junction,Pleasant Park,Dusty Depot,Salty Springs,Retail Row,Tomato Town,House on the Hill,Town west of Motel,Town northeast of Flush Factory,Buildings West of Tilted Towers,Buildings North of the Prison,Motel,Risky Reels,Mansion South of Lonely Lodge,Dusty Divot,Buildings East of Junk Junction,Silo in Mountain Northeast of Snobby Shores,Trailer Park"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-19 MuiSelect-icon-5"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="filterOption"
      >
        <label>
          GameType
        </label>
        <div
          className="MuiInput-root-6 MuiInput-underline-10"
        >
          <div
            className="MuiSelect-root-1"
          >
            <div
              aria-haspopup="true"
              aria-owns={null}
              aria-pressed="false"
              className="MuiSelect-select-2 MuiSelect-selectMenu-3 MuiInput-input-14"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              All
            </div>
            <input
              type="hidden"
              value="solo,duo,squad"
            />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-19 MuiSelect-icon-5"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="filterOption"
      >
        <div
          className="filterStatsOptionDropdown"
          onClick={[Function]}
        >
          Filter By Stats
        </div>
      </div>
    </div>
    <div
      className="submitGameButtonContainer"
    >
      <button
        className="submitGameButton"
        onClick={[Function]}
      >
        Filter!
      </button>
    </div>
  </div>
</div>
`;
